docs/content/concepts/configuration/config-schema.mdx

```diff
@@ -5,91 +5,119 @@ description: Job run configuration allows providing parameters to jobs at the ti
+#### Using software-defined assets
 
-During execution, the specified parameters are accessible within the body of the op/asset/resource under `context.op_config` (for ops/assets) or `context.resource_config` (for resources). It might seem confusing that asset config is accessed under `context.op_config` instead of `context.asset_config`. However, assets are wrappers for ops, so when we access asset config we are literally just accessing config for the underlying op.
+Here, we define a subclass of <PyObject object="Config"/> holding a single string value representing the name of a user. We can access the config through the `config` parameter in the asset body.
+
+```python file=/guides/dagster/pythonic_config/pythonic_config.py startafter=start_basic_asset_config endbefore=end_basic_asset_config dedent=4
+from dagster import asset, Config
+
+class MyAssetConfig(Config):
+    person_name: str
+
+@asset
+def greeting(config: MyAssetConfig) -> str:
+    return f"hello {config.person_name}"
+```
 
-Below we define a simple op and asset with identical `config_schemas` defining a single configurable parameter, `person_name`, as well as a resource with a configurable `url` parameter:
+</TabItem>
+<TabItem name="Using ops and jobs">
 
-```python file=/concepts/configuration/configurable_op_asset_resource.py startafter=start endbefore=end
-@op(config_schema={"person_name": str})
-def op_using_config(context):
-    return f'hello {context.op_config["person_name"]}'
+#### Using ops
 
+Here, we define a subclass of <PyObject object="Config"/> holding a single string value representing the name of a user. We can access the config through the `config` parameter in the op body.
 
-@asset(config_schema={"person_name": str})
-def asset_using_config(context):
-    # Note how asset config is also accessed with context.op_config
-    return f'hello {context.op_config["person_name"]}'
+```python file=/guides/dagster/pythonic_config/pythonic_config.py startafter=start_basic_op_config endbefore=end_basic_op_config dedent=4
+from dagster import op, Config
 
+class MyOpConfig(Config):
+    person_name: str
 
-@resource(config_schema={"url": str})
-def resource_using_config(context):
-    return MyDatabaseConnection(context.resource_config["url"])
+@op
+def print_greeting(config: MyOpConfig):
+    print(f"hello {config.person_name}")  # noqa: T201
 ```
 
-<Note>
-  It is technically possible to access <code>context.op_config</code> inside ops
-  (not assets) without defining a <code>config_schema</code>. However, this is
-  not recommended.
-</Note>
+</TabItem>
+</TabGroup>
 
 You can also build config into jobs, as described in [the Jobs documentation](/concepts/ops-jobs-graphs/jobs#configuring-jobs).
 
+### Defining and accessing Pythonic configuration for a resource
```

Comment:
Should this be its own section? Currently it's contained in the **Defining and accessing configuration for an op or asset** section, but it feels like it might belong on its own - thoughts?

Reply:
Yes, good catch